src/Uno.UI/UI/Xaml/Media/Imaging/WriteableBitmap.skia.cs

```diff
@@ -13,9 +13,7 @@ partial class WriteableBitmap
 
 		private protected override bool TryOpenSourceSync(int? targetWidth, int? targetHeight, out ImageData image)
 		{
-			_surface ??= new SkiaCompositionSurface();
-
-			_surface.CopyPixels(PixelWidth, PixelHeight, _buffer.AsReadOnlyMemory());
+			_surface = SkiaCompositionSurface.LoadFromPixels(PixelWidth, PixelHeight, _buffer.AsReadOnlyMemory());
```

Comment:
Previously, we were potentially re-using the same surface.

I think this change is not necessary for this PR and we can be better on the safe side and keep `CopyPixels` as instance method.

Reply:
No, it's necessary for exactly that reason. Using the same "surface" was wrong, since the SkiaCompositionSurface is just a wrapper for a simple SKImage and changing the image should cause render invalidation, which didn't previously happen because we were changing the internal SKImage without changing the surface instance.

Reply:
I don't see anything wrong with re-using the same surface. It just currently doesn't work the way it should.

@jeromelaban This is the root cause for the ProgressRing issue where we are reusing the same surface and not causing invalidate render. PR https://github.com/unoplatform/uno/pull/17059 has "what I think" the proper fix.

Reply:
The entire idea of a "surface" is misleading. It's not a surface, it's just a dumb wrapper for an SKImage because we can't implement `ICompositionSurface` directly on the SKImage. It doesn't make a lot of sense to "reuse the same surface" because there is no "surface". We would be "reusing the same wrapper but change the the thing is wraps", which is a lot of work to just "replace the thing". By making the wrapper a readonly struct, we make reasoning about the relationship between the wrapper and what it wraps a lot easier.

Reply:
As we discussed, https://github.com/unoplatform/uno/pull/15951/ will break by some of the changes in SkiaCompositionSurface. Let's wait for it to merge fix and then see what to do.